lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -395,12 +395,13 @@ int amf_idle_mode_procedure(amf_context_t* amf_ctx) {
       PARENT_STRUCT(amf_ctx, ue_m5gmm_context_s, amf_context);
   amf_ue_ngap_id_t ue_id = ue_context_p->amf_ue_ngap_id;
 
-  smf_context_t smf_ctx;
+  smf_context_t* smf_ctx = nullptr;
+  int j                  = 0;
 
   for (auto it = ue_context_p->amf_context.smf_ctxt_vector.begin();
-       it != ue_context_p->amf_context.smf_ctxt_vector.end(); it++) {
-    smf_ctx                   = *it;
-    smf_ctx.pdu_session_state = INACTIVE;
+       it != ue_context_p->amf_context.smf_ctxt_vector.end(); it++, j++) {
+    smf_ctx = ue_context_p->amf_context.smf_ctxt_vector.data() + j;
+    smf_ctx->pdu_session_state = INACTIVE;
```

Comment:
Looks quite convoluted. Why don't you simply use `it->pdu_session_state = INACTIVE`?